### PR TITLE
Docs: Add information about deprecation notices and deprecating features

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/vulcan/issues/584

## Proposed changes:

* This adds a markdown file which mainly focuses on explaining how to add deprecation notices.
* There is a small addition linking to information on how to deprecate code as well.
* This file is a WIP - to be expanded with more information about deprecating features.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/584

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Proof-read.

### CODE REVIEW

File: docs/deprecating-features.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+## Adding deprecation notices in Jetpack
+
+This refers to styled deprecation notices on specific admin pages, with custom calls to action, within the Jetpack plugin.
+
+In the [`Deprecate`](/projects/plugins/jetpack/src/class-deprecate.php) class within the Jetpack plugin, an array of notices exists within the constructor. By default this includes just one demo notice.
+
+In order to show a deprecation notice on WP Admin (dashboard only), Jetpack Settings and Dashboard page, as well as My Jetpack, you'll need to consider the following:
+
+* The `show` property is optional, but setting it to false will ensure the notice will not display anywhere.
+* The `hide_in_woa` property is also optional, and setting it to false will ensure the notice will not display on WoA sites.
```
Comment (fgiannar) [inline docs/deprecating-features.md]:
```suggestion
* The `hide_in_woa` property is also optional, and setting it to `true` will ensure the notice will not display on WoA sites.
```

**Review by fgiannar — COMMENTED** [review]
Many thanks for adding this, Karen!

We could mention that the deprecation notices are meant to be used on self-hosted and Atomic sites, but not for Simple ones.

Maybe we could further enrich this by adding a screenshot of the notice on WP Admin (dashboard), Jetpack settings and MyJetpack to highlight the different notices.

We could also be more verbose on the notice properties, aka mention also the required ones.

Finally, we could also include the demo notice here as an example before expanding on the required and non required properties.

Feel free to incorporate the suggestions that make sense to you, I only found one typo inline.

File: docs/deprecating-features.md (hunk 1 above)
Comment (coder-karen) [reply docs/deprecating-features.md]: Good catch.

**Comment by coder-karen** [comment]
Thanks for testing!

> Maybe we could further enrich this by adding a screenshot of the notice on WP Admin (dashboard), Jetpack settings and MyJetpack to highlight the different notices.

Yes I imagine we could (added now), I didn't realize we could add images to markdown in a repo but saw one on another markdown page when looking now.

I edited the file within the PR directly to add the images, perhaps there was a better way but that seemed to work. Clicking 'view file' here from the file viewer will show the images:

Screenshot 2024-11-07 at 13 04 27

I'm aware the styles will likely change when the notices are updated but we could make sure to update it here too.

As for the other suggestions, yes good idea to add in an example notice and expand on the properties here.

In doing so, I spotted an issue with the url code passed to Jetpack Settings and Dashboard notices (passed to the `DeprecationNotice` component) - it was missing `getRedirectUrl` to generate the support URL. I'll add this to a separate PR.

File: docs/deprecating-features.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+See the [coding-standards](/docs/coding-guidelines.md) document for more information about deprecating code.
+
+## Adding deprecation notices in Jetpack
+
+This refers to styled deprecation notices on specific admin pages, with custom calls to action, within the Jetpack plugin. These are meant for self-hosted and WoA sites, but not Simple sites (where notices won't show by default).
+
+In the [`Deprecate`](/projects/plugins/jetpack/src/class-deprecate.php) class within the Jetpack plugin, an array of notices exists within the constructor. By default this includes just one demo notice.
+
+In order to show a deprecation notice on WP Admin (dashboard only), Jetpack Settings and Dashboard page, as well as My Jetpack, you'll need to add to the `$notices` array in the `Deprecate` class. An demo already exists in the array.
```
Comment (bindlegirl) [inline docs/deprecating-features.md]: "A demo"?

Comment (bindlegirl) [reply docs/deprecating-features.md]: Other than that tiny thing, this looks good to me.

Comment (coder-karen) [reply docs/deprecating-features.md]: Good catch :) I expanded upon it a bit more as well in 76eca07811a47a6f4858f3287074961a6c1ca31c

**Review by fgiannar — APPROVED** [review]
Looks great! Again, many thanks for enriching our dev support docs!

**Pull request merged**